### ISSUE TITLE
TypeScript aria-* props should not be null

### ISSUE THREAD

**Issue by rynti** [jcoreio/material-ui-popup-state#22]
I just tested the latest version 1.5.0 of this package with its added TypeScript types. Unfortunately, I can't use it because my TypeScript compiler now gives me tons of warnings, since it seems that the aria-* props returned in the bindTrigger or other bind-methods might be null, and the Aria Props set by React are only allowed to be either undefined or their proper type (for example aria-describedby can only be a string or undefined, but not null).

Here is the [type of bindTrigger](https://github.com/jcoreio/material-ui-popup-state/blob/099a26ef12cc19ea60a4530a23fafd24693e9fea/core.d.ts#L58-L65) with null in their types:
```ts
export function bindTrigger(
  popupState: PopupState
): {
  'aria-owns'?: string | null | undefined
  'aria-describedby'?: string | null | undefined
  'aria-haspopup': true | null | undefined
  onClick: (event: SyntheticEvent<any>) => void
}
```

Here is an excerpt of the [AriaAttributes by @types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/02549c083e9d62b604091d770c4568d47282cdd0/types/react/index.d.ts#L1506):
```ts
// All the WAI-ARIA 1.1 attributes from https://www.w3.org/TR/wai-aria-1.1/
interface AriaAttributes {
    // ...
    'aria-describedby'?: string;
    // ...
}
```

I'm not sure how the type definitions were made for this project, but it seems the solution would be to just get rid of all the `| null` types for aria-* props.

I was using TypeScript version 3.7.3 when testing this.

**Comment by mjadczak** [comment]
I am seeing the same issue, but with the `id` prop when using `bindTrigger` and `bindMenu`:
```
 error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: MenuProps, context?: any): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | Component<...> | null', gave the following error.
    Type '{ children: Element[]; id: string | null | undefined; anchorEl: HTMLElement | null | undefined; open: boolean; onClose: () => void; onMouseLeave: (event: SyntheticEvent<any, Event>) => void; }' is not assignable to type 'MenuProps'.
      Types of property 'id' are incompatible.
        Type 'string | null | undefined' is not assignable to type 'string | undefined'.
          Type 'null' is not assignable to type 'string | undefined'.
  Overload 2 of 2, '(props: PropsWithChildren<MenuProps>, context?: any): ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<...>)> | Component<...> | null', gave the following error.
    Type '{ children: Element[]; id: string | null | undefined; anchorEl: HTMLElement | null | undefined; open: boolean; onClose: () => void; onMouseLeave: (event: SyntheticEvent<any, Event>) => void; }' is not assignable to type 'MenuProps'.
      Types of property 'id' are incompatible.
        Type 'string | null | undefined' is not assignable to type 'string | undefined'.
          Type 'null' is not assignable to type 'string | undefined'.

323           <Menu {...bindMenu(positionsPopupState)}>
```

Is the issue here that Flow is not as strict as TS in the difference between null and undefined? FWIW, I also have my own local TS types for this library, and I transcribed all the Flow maybe types as `T | undefined`.

**Comment by jedwards1211** [comment]
The way the TypeScript folks make it treat `null` and `undefined` differently is super dumb and wastes all of our time...thanks for letting me know, I'll fix it

**Comment by jedwards1211** [comment]
> Is the issue here that Flow is not as strict as TS in the difference between null and undefined
Yes, Flow doesn't foist piddly BS onto devs like this, though it does cause other serious headaches

**Comment by jedwards1211** [comment]
TS also deserves criticism for not allowing components more leeway in defining their own `children` property type:

```
test/menuRenderProps.test.tsx:8:4 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: Readonly<Props>): PopupState', gave the following error.
    Type '{ children: (string | ((popupState: PopupState) => Element))[]; variant: "popover"; popupId: string; }' is not assignable to type 'Readonly<Props>'.
      Types of property 'children' are incompatible.
        Type '(string | ((popupState: PopupState) => Element))[]' is not assignable to type '(props: PopupState) => ReactNode'.
          Type '(string | ((popupState: PopupState) => Element))[]' provides no match for the signature '(props: PopupState): ReactNode'.
  Overload 2 of 2, '(props: Props, context?: any): PopupState', gave the following error.
    Type '{ children: (string | ((popupState: PopupState) => Element))[]; variant: "popover"; popupId: string; }' is not assignable to type 'Readonly<Props>'.
      Types of property 'children' are incompatible.
        Type '(string | ((popupState: PopupState) => Element))[]' is not assignable to type '(props: PopupState) => ReactNode'.

8   <PopupState variant="popover" popupId="demoMenu">
     ~~~~~~~~~~



Found 1 error.
```

**Comment by jedwards1211** [comment]
@rynti @mjadczak I added some test .tsx files to make sure that there are no type errors in the build, but hopefully I didn't miss anything...if you have any other issues let me know and I'll reopen this!

**Comment by mjadczak** [comment]
> Yes, Flow doesn't foist this piddly BS onto devs, though it does cause other serious headaches

Unfortunately, the piddly BS comes from JavaScript itself, so I think caring about the distinction does make sense, annoying as it is.

With 1.5.1 my project compiles fine - thanks for fixing!

**Comment by rynti** [comment]
Can also confirm this is fixed with 1.5.1, thanks!

**Comment by jedwards1211** [comment]
@mjadczak yeah -- I wish no one would give `null` and `undefined` different meaning in APIs though, to me it's always asking for trouble.  In my own code I always try to treat them the same.
